### ISSUE TITLE
Enable focus state in `Checkbox` component when `disabled`

### ISSUE THREAD

**Issue by o-hook** [porsche-design-system/porsche-design-system#3726]
Scope
---
The `Checkbox` component sets an HTML `disabled` attribute on the `input` element when the `disabled` prop is set. This makes the component (and its state) not accessible to assistive technology users because it is not reachable. To be more compliant and consistent to other components (e.g. `Button`), only `aria-disabled` should be set while all interactions except `focus` are disabled.

Out of Scope
---

Notes  
---
- Should be the same functionality like the `Button` component.

Acceptance criteria
---
- [ ] `Checkbox` is still focusable while being in `disabled` state

Subtasks
---
- [ ] Task
